guides/tutorials/codelabs/kubernetes-v2-source-to-prod/index.md

```diff
@@ -0,0 +1,536 @@
+the __Clusters__ tab, and select the blue Deployment object attached to the
+Replica Set shown below:
+
+{% include figure
+   image_path="./staging-v001.png"
+%}
+
+We can see in the __Artifact__ section on the right that we have bound our
+Docker image as well as our ConfigMap.
+
+Let's see what our application is serving. Run:
+
+```bash
+kubectl proxy --context $STAGING_CONTEXT
+```
+
+And then visit [the sample
+service](http://localhost:8001/api/v1/proxy/namespaces/default/services/spinnaker-demo:80/)
+in your browser. Let's make a change to this service, and configure Spinnaker
+to listen to Docker builds.
+
+# 4. Configure Docker Triggers
+
+__Important__: We need to configure DockerHub to build on __Tag__ events only,
+if we build on every commit, this particular setup will trigger both when
+manifests & code are changed at once. This can be configured under your
+Docker repository's __Build Settings__ tab as shown here:
+
+{% include figure
+   image_path="./docker-tag-only.png"
+   caption="This build rule will create a matching image tag each time you push
+   a git tag."
+%}
+
+Next, in Spinnaker, let's edit our Pipeline to allow Docker images to trigger a
+deployment:
+
+First, add a Docker expected artifact next to our Git expected artifact:
+
+{% include figure
+   image_path="./configure-docker-artifact.png"
+%}
+
+Next, add a _Webhook_ trigger to listen to build events from DockerHub. The
+_Docker_ trigger alone won't provide us with provenance information.
+
+{% include figure
+   image_path="./configure-docker-webhook.png"
+%}
+
+Finally, back in the "Deploy (Manifest)" stage configuration, list that the
+Docker artifact is expected to bind in this deployment:
+
+{% include figure
+   image_path="./bind-docker.png"
+%}
+
+Save the pipeline.
+
+# 5. Deploy Docker to Staging
+
+You can push a tag to your repository by running:
+
+```bash
+git tag release-1.0
+git push origin release-1.0
+```
+
+{% include figure
+   image_path="./staging-webhook-execution.png"
+   caption="Notice that this time the trigger was a Webhook trigger, and we see
+   details about both types of artifacts that we deployed. Since the GitHub
+   file artifact was configure to __Use Prior Execution__, we redeployed the
+   same manifests as last time, but with a new Docker image. Because of this,
+   we did not deploy a new ConfigMap, and kept the version at `-v000`."
+%}
+
+> This deployment is a lot faster than the last one, since the docker image was
+> already pulled into our cluster, meaning it took less time for the images to
+> start running and appear as "Healthy".
+
+Back on the __Clusters__ tab we can see the deployment has rolled out our new
+image:
+
+{% include figure
+   image_path="./staging-v002.png"
+%}
+
+# 6. Manual Judgement
+
+For the sake of a simple codelab, we will gate deployments to production by a
+"Manual Judgement" pipeline. This can be replaced by a canary, integration
+test suite, or other mechanism for validating staging.
```

Comment:
"In practice" or "In a non-codelab situation." Something like that? I want to say "In production," but because we deploy to production in this codelab, that's confusing.

Worth saying you can also continue to use manual judgement?